apps/frontend/src/pages/CreateAccount.tsx

```diff
@@ -1,5 +1,201 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Button,
+  TextField,
+  Typography,
+  FormHelperText,
+  Link,
+} from "@mui/material";
+import martletPlaceLogo from "../images/martletplace-logo.png";
+import { colors } from "../styles/colors";
+import { useStyles } from "../styles/pageStyles";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+
 const CreateAccount = () => {
-  return <h1>CreateAccount</h1>;
+  const classes = useStyles();
+  const [email, setEmail] = useState("");
+  const [name, setName] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  const [emailError, setEmailError] = useState("");
+  const [usernameError, setUsernameError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+
+  const isFormIncomplete = !email || !name || !username || !password;
+
+  const navigate = useNavigate();
+
+  const handleCreateAccount = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // Check if email is a valid UVic email
+    if (!email.endsWith("@uvic.ca")) {
+      setEmailError("Please enter a valid UVic email.");
+      return;
+    }
+
+    // Regex for username validation
+    const usernameFormat = /^[a-zA-Z0-9]{1,20}$/;
+
+    if (!usernameFormat.test(username)) {
+      // Add or update an error state for username validation
+      setUsernameError(
+        "Username must be between 1 and 20 characters and only contain letters or numbers.",
+      );
+      return;
+    } else {
+      // Clear username error if valid
+      setUsernameError("");
+    }
+
+    if (
+      password.length >= 8 &&
+      /[a-z]/.test(password) &&
+      /[A-Z]/.test(password) &&
+      /[0-9]/.test(password) &&
+      /[\W_]/.test(password)
+    ) {
+      // Clear error message if password is valid
+      setPasswordError("");
+    } else {
+      setPasswordError("Password does not meet the requirements.");
+      return;
+    }
+
+    try {
```

Comment:
I believe we should be doing this with axios's own error catching as it should handle everything, but if there's a particular reason that you're doing it like this then don't let me slow down your progress! 
https://axios-http.com/docs/post_example

Reply:
Oh perfect, that looks so much better too! I missed this, thanks for catching it! 😁